src/syrupy/location.py

```diff
@@ -16,8 +17,8 @@ def __init__(self, node: Any):
 
     @property
     def classname(self) -> Optional[str]:
-        classes = self._node.obj.__qualname__.split(".")[:-1]
-        return ".".join(classes) if classes else None
+        _, __, qualname = self._node.location
```

Comment:
Where are you seeing documentation for "location"?

Reply:
ah found it. This is "Item" not "Node": https://docs.pytest.org/en/latest/_modules/_pytest/nodes.html#Item

Reply:
Traced the source code:

https://github.com/pytest-dev/pytest/blob/7d5f5a878598aaa7f3e100ef6a97bcd13c1b408d/src/_pytest/python.py#L328 -> https://github.com/pytest-dev/pytest/blob/7d5f5a878598aaa7f3e100ef6a97bcd13c1b408d/src/_pytest/python.py#L296

Reply:
I think it's `Function` which inherits from `Item` and for our use `PyobjMixin`
https://docs.pytest.org/en/latest/_modules/_pytest/python.html#Function

With the `location` property from `Node` using the `reportinfo` method defined in `PyobjMixin`

https://github.com/pytest-dev/pytest/blob/7d5f5a878598aaa7f3e100ef6a97bcd13c1b408d/src/_pytest/python.py#L315-L330